homeassistant/components/media_player/sonos.py

```diff
@@ -260,9 +286,14 @@ def __init__(self, hass, player):
         self._is_playing_tv = None
         self._favorite_sources = None
         self._source_name = None
-        self.soco_snapshot = Snapshot(self._player)
+        self._soco_snapshot = None
         self._snapshot_group = None
 
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Subscribe sonos events."""
+        self._subscribe_to_player_events()
```

Comment:
I look deeper into SoCo code and see that is not a good idea to run it inside loop. I move it into executor.